davinci/src/main/kotlin/com/pingidentity/davinci/collector/Device.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025 Ping Identity. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.pingidentity.davinci.collector
+
+import com.pingidentity.davinci.json
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.decodeFromJsonElement
+import kotlinx.serialization.json.jsonArray
+
+/**
+ * Data class representing a Device.
+ * @property id The id of the device.
+ * @property type The type of the device.
+ * @property title The title of the device.
+ * @property description The description of the device.
+ * @property iconSrc The icon source of the device.
+ * @property default Use this device as default.
+ * @property value The value of the device.
+ */
+@Serializable
+data class Device(
+    val id: String? = null,
```

Comment:
Why is the id optional? Is there a scenario that the device won't have an ID?

Reply:
from the array we get I do not see an ID, so are we supposed to create a UUID locally or we leave that blank?
`[{"type":"EMAIL","title":"Email","description":"Receive an authentication passcode in your email.","iconSrc":"https://assets.pingone.com/ux/end-user/2.10.0/images/icon-outline-mail.svg"},{"type":"SMS","title":"Text Message","description":"Receive an authentication passcode in a text message.","iconSrc":"https://assets.pingone.com/ux/end-user/2.18.2/images/icon-outline-text-message.svg"},{"type":"VOICE","title":"Voice","description":"Receive a phone call with an authentication passcode.","iconSrc":"https://assets.pingone.com/ux/end-user/2.10.0/images/icon-outline-call.svg"}]`

Reply:
The Device object share across Registration and Authentication, the Authentication contains additional attributes like **id**, **default**, **value**